common/dbutils/composite_keys.go

```diff
@@ -111,10 +112,10 @@ func GenerateStoragePrefix(addressHash common.Hash, incarnation uint64) []byte {
 	prefix := make([]byte, 0, common.HashLength+8)
 	prefix = append(prefix, addressHash[:]...)
 
-	//todo pool
-	buf := make([]byte, 8)
-	binary.BigEndian.PutUint64(buf, incarnation^^uint64(0))
-	prefix = append(prefix, buf...)
+	buf := pool.GetBuffer(8)
```

Comment:
Can avoid the buffer allocation altogether. Just do `binary.BigEndian.PutUint64` straight into the `prefix[common.Hash:]`